src/agent/onefuzz-agent/src/managed/cmd.rs

```diff
@@ -13,7 +19,17 @@ pub async fn run(args: &clap::ArgMatches<'_>) -> Result<()> {
     let config = Config::from_file(config_path, setup_dir)?;
 
     init_telemetry(config.common());
-    let result = config.run().await;
+
+    let result = tokio::select! {
+        result = config.run() => result,
+
+        // Ignore this task if it returns due to a querying error.
+        Ok(oom) = out_of_memory(MIN_AVAILABLE_BYTES) => {
```

Comment:
@chkeita, note how this differs a bit from our earlier discussion.

The supervisor `Agent` is implemented using (dynamic trait object) indirection to allow integration testing with test doubles for its internal components. Trying to add this global resource check as-is would pollute that, and was actually a little awkward (the supervisor really needs to be refactored to be `select`-driven).

This was the most minimal alternative way I could think up to get the behavior change I wanted, which was: "if we are in the `busy` state, running a task, and we're about to OOM, stop the task with an informative error".